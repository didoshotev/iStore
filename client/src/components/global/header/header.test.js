import React from 'react'
import Header from './header'
import renderer from 'react-test-renderer'
import TestingEnvironment from '../../../test-utils/router'
import getNavigation from '../../../utils/navigation'

describe('Header Component', () => {
    it('return auth routes from getNavigation', () => {
        const links = getNavigation({
            loggedIn: true,
        })

        const authLinks = [
            {
                title: 'All Products',
                link: '/products'
            },
            {
                title: 'Mac',
                link: '/mac'
            },
            {
                title: 'iPad',
                link: '/ipad'
            },
            {
                title: 'iPhone',
                link: '/iphone'
            },
            {
                title: 'Logout',
                link: '/logout'
            },
            {
                title: 'Cart',
                link: `/cart`
            }
        ]

        expect(links).toStrictEqual(authLinks)
    })

    it('should render auth routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    loggedIn: true,
                }
            }}>
                <Header />
            </TestingEnvironment>

        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('should render non-auth routes', () => {
        const tree = renderer.create(
            <TestingEnvironment value={{
                user: {
                    loggedIn: false,
                }
            }}>
                <Header />
            </TestingEnvironment>

        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})