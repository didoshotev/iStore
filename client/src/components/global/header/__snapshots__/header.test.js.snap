// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render auth routes 1`] = `
<header
  className="header-background"
>
  <ul
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="logo"
        src="https://istyle.bg/media/logo/stores/11/iSTYLE-logo-white-red.svg"
      />
    </a>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/products"
        onClick={[Function]}
      >
        All Products
      </a>
    </li>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/mac"
        onClick={[Function]}
      >
        Mac
      </a>
    </li>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/ipad"
        onClick={[Function]}
      >
        iPad
      </a>
    </li>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/iphone"
        onClick={[Function]}
      >
        iPhone
      </a>
    </li>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
  </ul>
</header>
`;

exports[`Header Component should render non-auth routes 1`] = `
<header
  className="header-background"
>
  <ul
    className="header"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="logo"
        src="https://istyle.bg/media/logo/stores/11/iSTYLE-logo-white-red.svg"
      />
    </a>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/products"
        onClick={[Function]}
      >
        All Products
      </a>
    </li>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/mac"
        onClick={[Function]}
      >
        Mac
      </a>
    </li>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/ipad"
        onClick={[Function]}
      >
        iPad
      </a>
    </li>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/iphone"
        onClick={[Function]}
      >
        iPhone
      </a>
    </li>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
    <li
      className="header-li"
    >
      <a
        className="a a--white"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
  </ul>
</header>
`;
